libkpod/stats.go

```diff
@@ -0,0 +1,161 @@
+package libkpod
+
+import (
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer"
+)
+
+// ContainerStats contains the statistics information for a running container
+type ContainerStats struct {
+	Container   string
+	CPU         float64
+	cpuNano     uint64
+	systemNano  uint64
+	MemUsage    uint64
+	MemLimit    uint64
+	MemPerc     float64
+	NetInput    uint64
+	NetOutput   uint64
+	BlockInput  uint64
+	BlockOutput uint64
+	PIDs        uint64
+}
+
+// GetContainerStatsFollow gets the running stats for a running container continually
+func (c *ContainerServer) GetContainerStatsFollow(ids []string, all bool, statsChan chan []*ContainerStats, times int) error {
```

Comment:
Eventually this will be a method of a single container, so I'd greatly prefer the API call also act on a single container. I don't think it's necessary to requery every iteration of the loop - it should be sufficient to query for running containers, call GetContainerStats on all of them, and print the output. If some the containers are shut down, we can ignore the error and not print it. We then requery for live containers before running the loop again. There is a small race where a container can be started after we query but before we print, but that's a very small race and the container will show up in the next set of output.